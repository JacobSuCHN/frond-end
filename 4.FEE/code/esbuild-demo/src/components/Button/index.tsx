import React from "react";

const buttonTypes = ["default", "primary", "secondary"] as const;

type ButtonType = (typeof buttonTypes)[number];

interface ButtonProps extends Omit<HTMLButtonElement, "type"> {
  type?: ButtonType;
}

export const Button: React.FC<ButtonProps> = ({
  type = "default",
  children,
}) => {
  return <button className={`button button-${type}`}>{children}</button>;
};
