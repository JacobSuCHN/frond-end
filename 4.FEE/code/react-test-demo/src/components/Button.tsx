import React from "react";
import { useBoolean } from "../hooks/useBoolean";

const buttonTypes = ["primary", "secondary", "danger"] as const;

type buttonTypes = (typeof buttonTypes)[number];

interface IButtonProps {
  type?: buttonTypes;
  children: React.ReactNode;
}

export const Button: React.FC<IButtonProps> = ({ type, children }) => {
  const [isDisabled, toggle] = useBoolean(false);
  return (
    <button disabled={isDisabled} onClick={toggle}>
      {children}
    </button>
  );
};
