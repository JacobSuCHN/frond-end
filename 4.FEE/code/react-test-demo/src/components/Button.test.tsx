import React from "react";
import { describe, expect, it } from "@jest/globals";
import renderer from "react-test-renderer";
import { Button } from "./Button";

describe("button test", () => {
  it("should render", () => {
    const component = renderer.create(<Button>button</Button>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
