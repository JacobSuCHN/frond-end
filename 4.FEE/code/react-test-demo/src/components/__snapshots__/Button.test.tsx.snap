// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button test should render 1`] = `
<button
  disabled={false}
  onClick={[Function]}
>
  button
</button>
`;
